sync/upstream.py

```diff
@@ -687,9 +687,8 @@ def updates_for_backout(git_gecko: Repo,
             sync = syncs.pop()
             assert isinstance(sync, UpstreamSync)
             if commit in sync.gecko_commits:
-                # This commit was already processed
-                backed_out_commit_shas = set()
-                return {}, {}
+                # This commit was already processed for this sync
+                continue
```

Comment:
yeah, you're right about the removing from `backed_out_commit_shas`, I've updated this.
Regarding the scenario when S1 is merged, I guess I'm missing something, but shouldn't S1 then have status `complete` and because of that excluded from all the backout updates, so it shouldn't come to the "create a sync" point?

Reply:
It's exactly because it's merged that we get to the create a sync point. The whole "create a new sync for the backout commit" is designed to handle the case where e.g. something is backed out of central _after_ the corresponding upstream sync has already been merged. In that case we can't do anything with the original PR (because GitHub quite rightly doesn't allow adding new commits to a merged PR) so instead we want to create an entirely new sync/PR for the backout.

Reply:
Ok, I think I'm up to speed with what you're saying :)
Just to reiterate, basically the problem is that in the case you described, we're creating a new sync with only the backout commit, which works fine for one bug backout, but doesn't for multiple bugs, where we should revert only the commits, which belong to the bug, which was merged already.
But so far I don't have a better idea, than, as you said, creating a sync with reverted commits for the bug. So, I guess, I'll try to sketch something.